### ISSUE TITLE
Setting styles and values - order of execution

### ISSUE THREAD

**Issue by ivanfemia** [abap2xlsx/abap2xlsx#150]
Currently the order in which a cells value and style are set influences the output.
Example:

<pre><code>
  lo_style                    = lo_excel->add_new_style( ).
  lo_style->font->bold        = abap_true.
  lo_style->font->italic      = abap_true.
  lo_style->font->name        = zcl_excel_style_font=>c_name_arial.
  lo_style->font->scheme      = zcl_excel_style_font=>c_scheme_none.
  lo_style->font->color-rgb   = zcl_excel_style_color=>c_red.
  lo_style_guid               = lo_style->get_guid( ).

  lo_worksheet->set_cell(       ip_column = 'B' ip_row = 2 ip_value = 'Cellvalue and Style set in method set_cell' ip_style = lo_style_guid ).

  lo_worksheet->set_cell(       ip_column = 'B' ip_row = 3 ip_value = 'Cellvalue set in method set_cell, style set afterwards' ).
  lo_worksheet->set_cell_style( ip_column = 'B' ip_row = 3 ip_style = lo_style_guid ).

  lo_worksheet->set_cell_style( ip_column = 'B' ip_row = 4 ip_style = lo_style_guid ).
  lo_worksheet->set_cell(       ip_column = 'B' ip_row = 4 ip_value = 'Cellstyle set in method set_cell_style, cellvalue set afterwards' ).
</code></pre>

The output in rows 2 and 3 is identical as expected, but the output in row 4 is different ( it does not use the style the cell was set to but the default style ).
This is counter intuitive and should be changed to have the same output for all 3 rows.

**Comment by ivanfemia** [comment]
I changed the method "SET_CELL" insofar as when no style is supplied in the method, the style that is currently used by the cell will be used instead.
This should solve the issue and the order of execution of set value and set style should not matter any more.

When changing the method I found some coding that set a standard-dateformat and standard-timeformat if no style was supplied.  I built a quick testreport and saw that this currently leads to a strange effect.  If you enter a date-field, the field will only be displayed as a datefield in EXCEL if the number-format was submitted with the style or if no style was supplied at all.
Thus if you created a style say "bold and red" without any numberformatting you were not able to display dates correctly but all (character) fields.
I believe that an abap2xlsx-user will expect all fields to be displayed in the style he passes and in a correct format - especially if he set the default-dateformat with the corresponding method.  To ensure this I moved the coding to the end of the set_cell-method and modified it insofar as it will now try to display a date as such whenever no numberformatting is supplied instead of whenever no style was submitted.

Last and least I changed two shift-statements that were used to eliminate a trailing blank  into one condense-statement.  A small testprogram showed this to be faster in my working system.  Maybe someone could try this in a different system to confirm this - otherwise I'd reset this last change.
